src/zenml/cli/stack.py

```diff
@@ -419,26 +420,39 @@ def register_stack(
 
                 if component_selected is None:
                     if service_connector_resource_model is None:
-                        if isinstance(service_connector, UUID):
-                            service_connector_resource_model = (
-                                client.verify_service_connector(
-                                    service_connector
+                        with console.status(
+                            "Exploring resources available to the service connector...\n"
+                        ):
+                            if isinstance(service_connector, UUID):
+                                service_connector_resource_model = (
+                                    client.verify_service_connector(
+                                        service_connector, timeout=120
+                                    )
                                 )
-                            )
-                        else:
-                            _, service_connector_resource_model = (
-                                client.create_service_connector(
-                                    name=stack_name,
-                                    connector_type=service_connector.type,
-                                    auth_method=service_connector.auth_method,
-                                    configuration=service_connector.configuration,
-                                    register=False,
+                                existing_service_connector_info = (
+                                    client.get_service_connector(
+                                        service_connector
+                                    )
                                 )
-                            )
-                            if service_connector_resource_model is None:
-                                cli_utils.error(
-                                    f"Failed to validate service connector {service_connector}..."
+                                generate_temporary_tokens = existing_service_connector_info.configuration.get(
+                                    "generate_temporary_tokens", True
                                 )
+                            else:
+                                _, service_connector_resource_model = (
+                                    client.create_service_connector(
+                                        name=stack_name,
+                                        connector_type=service_connector.type,
+                                        auth_method=service_connector.auth_method,
+                                        configuration=service_connector.configuration,
+                                        register=False,
+                                        timeout=120,
+                                    )
+                                )
+                                generate_temporary_tokens = False
```

Comment:
I might be mistaken, but shouldn't this be initialized to True ?

Reply:
Maybe the naming is not perfect, it is more a flag for the ability to use this flag as False, which is needed for Skypilot. I'll look into renaming.